pkgs/tools/networking/curl/default.nix

```
@@ -177,16 +180,22 @@ stdenv.mkDerivation rec {
     ln $out/lib/libcurl${stdenv.hostPlatform.extensions.sharedLibrary} $out/lib/libcurl-gnutls${stdenv.hostPlatform.extensions.sharedLibrary}.4.4.0
   '';
 
-  passthru = {
+  passthru = let
+    useThisCurl = attr: attr.override { curl = finalAttrs.finalPackage; };
+  in {
     inherit opensslSupport openssl;
     tests = {
-      inherit curlpp coeurl;
-      haskell-curl = haskellPackages.curl;
-      ocaml-curly = ocamlPackages.curly;
-      php-curl = phpExtensions.curl;
-      pycurl = python3.pkgs.pycurl;
+      curlWithTests = finalAttrs.finalPackage.overrideAttrs (_: { doCheck = true; });
```

```suggestion
      withCheck = finalAttrs.finalPackage.overrideAttrs (_: { doCheck = true; });
```

So two things
 - `curl` is a bit redundant. `withTests` might be more meaningful to other code.
 - `tests` -> `check`, as we're setting `doCheck`